Fix TreePathConverter by using String parsing

melezov opened the pull request:
Previous implementation couldn't read labels longer then the temporary buffer size.
Make tests compile - JsonWriter(byte[]) constructor is no longer visible

Commit:
Fix TreePathConverter by using String parsing

Previous implementation couldn't read labels longer then the temporary buffer size.
Make tests compile - JsonWriter(byte[]) constructor is no longer visible

zapov commented:
Use new JsonWriter(null) to avoid using deperecated API (or JsonWriter(byte[], null)
Error message must include position in JSON where the error happened.

Commit:
Fix TreePath.checkParts to only accept [a-zA-Z-0-9] pattern
Moved json tests to com.dslplatform.json package
Add reader.positionInStream to TreePathConverter Exception message

melezov commented:
I've moved JSON tests into com.dslplatform.json package, as those constructors are not public.
Perhaps they should be.
Fixed TreePath check, as it did not dismiss Unicode "letters".
Don't know about the position error, why isn't the nested exception good enough?
